File: php/ApiClient.php
<?php

class ApiClient
{
    private $appId;
    private $appSecret;
    private $baseUrl;

    public function __construct(string $appId, string $appSecret, string $baseUrl)
    {
        $this->appId = $appId;
        $this->appSecret = $appSecret;
        $this->baseUrl = rtrim($baseUrl, '/'); // 移除多余斜杠
    }

    public function make(string $url): array
    {
        $endpoint = '/67d27c333e192';
        return $this->sendGetRequest($endpoint, ['url' => $url]);
    }

    private function sendGetRequest(string $endpoint, array $params): array
    {
        // 添加 app_id 参数
        $data = $params;
        $data['app_id'] = $this->appId;

        // 签名
        $sign = $this->getSign($data);
        ksort($data);
        // 构建查询参数
        $queryString = http_build_query($data);
        // 构建完整 URL
        $url = $this->baseUrl . $endpoint . '?' . $queryString;
        // 创建请求上下文（带上 Sign 头）
        $context = stream_context_create([
            'http' => [
                'method' => 'GET',
                'header' => "Sign: {$sign}\r\n",
                'timeout' => 60,
            ]
        ]);

        // 发送请求
        $response = @file_get_contents($url, false, $context);
        if ($response === false) {
            throw new RuntimeException("API 请求失败: $url");
        }

        // 解析 JSON
        $result = json_decode($response, true);

        if (json_last_error() !== JSON_ERROR_NONE) {
            throw new RuntimeException("JSON 解析失败: " . json_last_error_msg());
        }

        return $result;
    }

    private function getSign(array $data): string
    {
        // 1. 按 key 升序排序
        ksort($data);

        // 2. 构造 key=value& 拼接字符串（仅对 value 进行 urlencode）
        $parts = [];
        foreach ($data as $key => $value) {
            $encodedValue = urlencode($value);
            $parts[] = "{$key}={$encodedValue}";
        }

        $preStr = implode('&', $parts);

        // 3. 拼接 appSecret
        $preStr .= $this->appSecret;
        // 4. 返回 MD5 签名
        return md5($preStr);
    }
}
$api = new ApiClient("appId", "appSecret", "https://gateway.diadi.cn/api");
print($api->make("url"));

File: python/ApiClient.py
import hashlib
import json
import urllib.parse
import urllib.request
import ssl


class ApiClient:
    def __init__(self, app_id: str, app_secret: str, base_url: str):
        self.app_id = app_id
        self.app_secret = app_secret
        self.base_url = base_url.rstrip('/')

    def make(self, url: str) -> dict:
        endpoint = '/parse'
        return self.send_get_request(endpoint, {'url': url})

    def send_get_request(self, endpoint: str, params: dict) -> dict:
        # 添加 app_id 参数
        data = params.copy()
        data['app_id'] = self.app_id

        # 签名
        sign = self.get_sign(data)

        # 排序后构建 query string
        sorted_data = dict(sorted(data.items()))
        query_string = urllib.parse.urlencode(sorted_data)

        # 构建完整 URL
        full_url = f"{self.base_url}{endpoint}?{query_string}"
        # 构建请求对象，带 Sign 头
        req = urllib.request.Request(full_url)
        req.add_header('Sign', sign)

        # 忽略 SSL 验证
        context = ssl._create_unverified_context()
      
        try:
            with urllib.request.urlopen(req, timeout=60, context=context) as response:
                body = response.read().decode('utf-8')
                result = json.loads(body)
                return result
        except Exception as e:
            print(f"请求失败: {e}")
            return {}

    def get_sign(self, data: dict) -> str:
        # 1. 按 key 升序排序
        sorted_items = sorted(data.items())

        # 2. 拼接 key=value（urlencode value）字符串
        parts = [f"{key}={urllib.parse.quote_plus(str(value))}" for key, value in sorted_items]
        pre_str = '&'.join(parts)

        # 3. 拼接 appSecret
        pre_str += self.app_secret

        # 4. 返回 MD5 签名
        return hashlib.md5(pre_str.encode('utf-8')).hexdigest()


# 使用示例
api = ApiClient("", "", "https://gateway.diadi.cn/api")
print(api.make("url"))
